yourbench/pipeline/summarization.py

```diff
@@ -0,0 +1,497 @@
+        error occurs or if inputs are empty.
+    """
+    try:
+        safe_preds = [p if p else "" for p in predictions]
+        safe_refs = [r if r else "" for r in references]
+
+        if not safe_preds or not safe_refs:
+            logger.warning("Skipping METEOR due to empty inputs.")
+            return 0.0
+
+        meteor_metric = evaluate.load("meteor")
+        meteor_result = meteor_metric.compute(predictions=safe_preds, references=safe_refs)
+        return meteor_result.get("meteor", 0.0)
+    except Exception as e:
+        logger.error("Error computing METEOR score: {}", str(e))
+        return 0.0
+
+
+def _safe_compute_rouge_scores(predictions: List[str], references: List[str]) -> Dict[str, float]:
+    """
+    Compute ROUGE scores safely with error handling.
+
+    Args:
+        predictions (List[str]): Candidate summaries or predicted texts.
+        references (List[str]): Reference texts matching each prediction.
+
+    Returns:
+        Dict[str, float]: Contains 'rouge1', 'rouge2', and 'rougeL' scores.
+        Returns a dictionary of zeros if an error occurs or if inputs are empty.
+    """
+    try:
+        safe_preds = [p if p else "" for p in predictions]
+        safe_refs = [r if r else "" for r in references]
+
+        if not safe_preds or not safe_refs:
+            logger.warning("Skipping ROUGE due to empty inputs.")
+            return {"rouge1": 0.0, "rouge2": 0.0, "rougeL": 0.0}
+
+        rouge_metric = evaluate.load("rouge")
+        rouge_result = rouge_metric.compute(predictions=safe_preds, references=safe_refs)
+        return {
+            "rouge1": rouge_result.get("rouge1", 0.0),
+            "rouge2": rouge_result.get("rouge2", 0.0),
+            "rougeL": rouge_result.get("rougeL", 0.0),
+        }
+    except Exception as e:
+        logger.error("Error computing ROUGE scores: {}", str(e))
+        return {"rouge1": 0.0, "rouge2": 0.0, "rougeL": 0.0}
+
+
+def _safe_compute_bert_score_f1(predictions: List[str], references: List[str]) -> float:
+    """
+    Compute average BERTScore (F1) safely with error handling.
+
+    Args:
+        predictions (List[str]): Candidate summaries or predicted texts.
+        references (List[str]): Reference texts matching each prediction.
+
+    Returns:
+        float: The average BERTScore-F1. Returns 0.0 if an error occurs or inputs
+        are empty.
+    """
+    try:
+        safe_preds = [p if p else "" for p in predictions]
+        safe_refs = [r if r else "" for r in references]
+
+        if not safe_preds or not safe_refs:
+            logger.warning("Skipping BERTScore due to empty inputs.")
+            return 0.0
+
+        bert_score_metric = evaluate.load("bertscore")
+        bert_result = bert_score_metric.compute(
+            predictions=safe_preds, references=safe_refs, model_type="bert-base-uncased"
+        )
+        f1_scores = bert_result.get("f1", [])
+        if not f1_scores:
```

Comment:
You would likely want to print a warning if we reach this step

Reply:
removed metric calculations in general.